apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/ShowTokenActions/TopTokenActions.tsx

```diff
@@ -1,36 +1,61 @@
-import {useSwap, useSwapPoolsByPair} from '@yoroi/swap'
+import {getBuyAmountbyChangingSell, useSwap, useSwapPoolsByPair} from '@yoroi/swap'
 import React from 'react'
 import {StyleSheet, View} from 'react-native'
 import {TouchableOpacity} from 'react-native-gesture-handler'
 
 import {Icon} from '../../../../../../components'
 import {LoadingOverlay} from '../../../../../../components/LoadingOverlay'
 import {COLORS} from '../../../../../../theme'
+import {asQuantity} from '../../../../../../yoroi-wallets/utils'
 import {ButtonGroup} from '../../../../common/ButtonGroup/ButtonGroup'
 import {useStrings} from '../../../../common/strings'
 import {useSwapTouched} from '../../../../common/SwapFormProvider'
 
 export const TopTokenActions = () => {
   const strings = useStrings()
   const orderTypeLabels = [strings.marketButton, strings.limitButton]
-
-  const {createOrder, orderTypeChanged} = useSwap()
-
+  const {createOrder, orderTypeChanged, selectedPoolChanged, limitPriceChanged, buyAmountChanged} = useSwap()
   const {isBuyTouched, isSellTouched} = useSwapTouched()
   const isDisabled = !isBuyTouched || !isSellTouched || createOrder.selectedPool === undefined
-
   const orderTypeIndex = createOrder.type === 'market' ? 0 : 1
-  const handleSelectOrderType = (index: number) => {
-    orderTypeChanged(index === 0 ? 'market' : 'limit')
-  }
 
-  const {refetch, isLoading} = useSwapPoolsByPair({
+  const {refetch, isLoading, poolList} = useSwapPoolsByPair({
     tokenA: createOrder.amounts.sell.tokenId ?? '',
     tokenB: createOrder.amounts.buy.tokenId ?? '',
   })
 
+  const handleSelectOrderType = (index: number) => {
+    if (index === 0) {
+      handleSync()
+    } else {
+      orderTypeChanged('limit')
+    }
+  }
+
   const handleSync = () => {
+    orderTypeChanged('market')
     refetch()
+    if (poolList !== undefined) {
+      const bestPool = poolList.map((a) => a).sort((a, b) => a.price - b.price)[0]
```

Comment:
avoid index accessing -> `.find(() => true)` | `const [bestPool, _] = poolList.` it also looks like this is a helper for the package `findPoolByBestPrice(poolList)`